dump/pldm_oem_cmds.cpp

```diff
@@ -73,11 +71,13 @@ void newFileAvailable(uint32_t dumpId, pldm_fileio_file_type pldmDumpType,
                     pldmInstanceId, dumpId, static_cast<uint16_t>(pldmDumpType),
                     dumpSize, newFileAvailReqMsg.size())
             .c_str());
+
     int retCode = encode_new_file_req(
         pldmInstanceId, pldmDumpType, dumpId, dumpSize,
         reinterpret_cast<pldm_msg*>(newFileAvailReqMsg.data()));
     if (retCode != PLDM_SUCCESS)
     {
+        freePLDMInstanceID(pldmInstanceId, mctpEndPointId);
```

Comment:
RAII, enclose pldminstance id and mctp endpoint id in an internal c++ class object and when that goes out of scope it will call freePLDMInstanceID